cmake/Modules/LocalRav1e.cmake

```diff
@@ -97,19 +97,10 @@ else()
     file(MAKE_DIRECTORY ${RAV1E_INCLUDE_DIR})
     set(RAV1E_FOUND ON)
 
-    set(RAV1E_LIBRARIES ${Rust_CARGO_TARGET_LINK_NATIVE_LIBS})
-    if(WIN32)
-        # If an item starts with "/" (e.g., "/defaultlib:msvcrt"), change the
-        # leading "/" to "-" so that the target_link_libraries() call below
-        # will treat the item as a linker flag.
-        list(TRANSFORM RAV1E_LIBRARIES REPLACE "^/" "-")
-        # Remove msvcrt from RAV1E_LIBRARIES since it's linked by default
-        list(REMOVE_ITEM RAV1E_LIBRARIES "msvcrt.lib" "-lmsvcrt")
-    endif()
-
     add_library(rav1e::rav1e STATIC IMPORTED)
     add_dependencies(rav1e::rav1e rav1e)
-    target_link_libraries(rav1e::rav1e INTERFACE "${RAV1E_LIBRARIES}")
+    target_link_libraries(rav1e::rav1e INTERFACE "${Rust_CARGO_TARGET_LINK_NATIVE_LIBS}")
+    target_link_options(rav1e::rav1e INTERFACE "${Rust_CARGO_TARGET_LINK_OPTIONS}")
```

Comment:
I am not sure if we should use `INTERFACE` or `PRIVATE` here.

Reply:
INTERFACE is the way to go because we are creating a library from scratch. Setting PRIVATE dependencies would not change anything as it is a shell library. INTERFACE ensures anybody depending on rav1e:rav1e will also use the dep.

Reply:
Since `Rust_CARGO_TARGET_LINK_OPTIONS` is an empty list on non-Windows platforms, I wonder if we should guard the `target_link_options()` call in an if statement:

```
    if (Rust_CARGO_TARGET_LINK_OPTIONS)
        target_link_options(rav1e::rav1e INTERFACE "${Rust_CARGO_TARGET_LINK_OPTIONS}")
    endif()
```

I am also not sure if we need to quote `${Rust_CARGO_TARGET_LINK_NATIVE_LIBS}` and `${Rust_CARGO_TARGET_LINK_OPTIONS}` in the `target_link_libraries()` and `target_link_options()` calls.

Reply:
> Since `Rust_CARGO_TARGET_LINK_OPTIONS` is an empty list on non-Windows platforms, I wonder if we should guard the `target_link_options()` call in an if statement:
> 
> ```
>     if (Rust_CARGO_TARGET_LINK_OPTIONS)
>         target_link_options(rav1e::rav1e INTERFACE "${Rust_CARGO_TARGET_LINK_OPTIONS}")
>     endif()
> ```
> 

That seems fine.

> I am also not sure if we need to quote `${Rust_CARGO_TARGET_LINK_NATIVE_LIBS}` and `${Rust_CARGO_TARGET_LINK_OPTIONS}` in the `target_link_libraries()` and `target_link_options()` calls.

Probably not given it would be a single item or ';' separated list. Though I think quoting is usually safest.

Reply:
Thanks for the reply. I did some experiments and found that both of the following are harmless:

```
add_library(wtc wtc.c)
target_link_options(wtc INTERFACE)
target_link_options(wtc INTERFACE "")
```

So we can just call `target_link_options()` unconditionally here.